processmanager/processinfo.go

```diff
@@ -591,20 +591,12 @@ func (pm *ProcessManager) SynchronizeProcess(pr process.Process) {
 		return
 	}
 	if len(mappings) == 0 {
-		// Valid process without any (executable) mappings. All cases are
-		// handled as process exit. Possible causes and reasoning:
-		// 1. It is a kernel worker process. The eBPF does not send events from these,
-		//    but we can see kernel threads here during startup when tracer walks
-		//    /proc and tries to synchronize all PIDs it sees.
-		//    The PID should not exist anywhere, but we can still double check and
-		//    make sure the PID is not tracked.
-		// 2. It is a normal process executing, but we just sampled it when the kernel
-		//    execve() is rebuilding the mappings and nothing is currently mapped.
-		//    In this case we can handle it as process exit because everything about
-		//    the process is changing: all mappings, comm, etc. If execve fails, we
-		//    reaped it early. If execve succeeds, we will get new synchronization
-		//    request soon, and handle it as a new process event.
-		pm.processPIDExit(pid)
+		// TODO: Check if main thread has exited, e.g. /proc/PID/stat will show zombie
+		log.Warnf("%v: main thread exit", pid)
```

Comment:
Regardless, we can play it safe and also keep the previous logic if we add a check here to detect with certainty that the main thread has exited. Right now, the code assumes that this will be the case whenever `len(mappings) == 0`.